### DISCUSSION TITLE
Presenting an open ended response later in the study

### DISCUSSION THREAD

**Discussion by Deewa-A** [jspsych/jsPsych#1433] [Q&A]
Hi,
I am trying to see if there is a way to include a participants open ended response to a question (completed during the middle of the study) and place it into a sentence later (end of the study).

What I am trying to do here is to draw upon participants response about their identity and include their response to that question in the middle of later sentence (i.e. Earlier you mentioned you had a ________ disability...). Is this feasible with jsPsych?

I am fairly new to jsPsych and coding, so pardon me if this question is unclear.

**Comment by becky-gilbert** [comment]
Yes, you can do this using jsPsych's [dynamic parameters](https://www.jspsych.org/overview/trial/#dynamic-parameters) for the later sentence stimulus. Dynamic parameters allow you to use a function to define/return the parameter value _during_ the experiment, right before the trial begins.

In your case, you'll need a way of finding the response from the earlier question in the jsPsych data. To do this, I suggest looking at the [documentation for data manipulation](https://www.jspsych.org/overview/data/#aggregating-and-manipulating-jspsych-data) (specifically the [data filtering reference](https://www.jspsych.org/core_library/jspsych-data/#filter) and examples).

When you put these things together, your later trial that includes their response will look something like this (note that I haven't tested this - it may not work!):
```javascript
var later_trial = {
  type: 'html-keyboard-response',
  stimulus: function() {
    // get data from the earlier trial
    var previous_trial = jsPsych.data.get().filter({})[0]; // <-- add criteria to the filter function to find the earlier trial data
    // extract the response from the data
    var previous_response = JSON.parse(previous_trial.responses[0]);
    // construct the stimulus string
    var stim = 'Earlier you mentioned you had a ';
    stim += previous_response;
    stim += "disability...";
    return stim;
  }
};
```
Feel free to follow up here and post your code if you need further guidance.
